include/addendum.txt

```diff
@@ -0,0 +1,73 @@
+`Gen3 Butler`_, as well as the `SuperTask`_ to create a **quantum graph**
+(workflow like graph) that can be executed to run the actual pipelines. This
+work is in the implementation phase, depending on the results and changes of
+this Gen3 Middleware we will re-evaluate some of the workflow systems to see
+which on fits best in this framework.
+
+During this time we have briefly looked at the Parsl software as another option
+of a workflow system as well as take the DESDM framework, which we have used in
+operations already, and ported it to Python 3 to be ready as another solution.
+
+DESDM Framework
+---------------
+
+In the past year we have looked at the DESDM framework as one of the options to
+execute the pipelines in an operations environment.  At NCSA we have experience
+running this as part of the DES project running in production for the past 5
+years for nightly processing as well as data releases.  NCSA has trained
+operators to run the software, analyze the results and retry select sections of
+the workflow. The system is ready to use and can execute tasks in parallel on a
+HTCondor cluster. Code is available for glide-ins, allowing an existing cluster
+(such as **lsst-dev**) to be configured to allow for execution of tasks from the
+DESDM framework. The framework will collect statistics, enabling us to do
+performance testing, as well as provenance tracking allowing us to find out
+exactly for each data file generated how it was generated, with what inputs and
+by whom. In the past months we have spend time on converting the existing code
+base from python 2 to python 3 to fit into the LSST software stack.  We also
+wrote LSST-specific plugins to enable running of the Gen2 LSST pipelines.   We
+have done tests using both the Python 2 version running the LSST software stack
+and RC dataset as well as using the python 3 version and the same dataset to
+test whether the DESDM framework is ready to run the existing LSST pipelines.
+
+Parsl
+-----
+
+`Parsl`_ (Parallel Scripting Library) is scripting library written in Python.
+The current version is 0.5.1. Parsl has been under development for the past 2
+years and is developed as a collaboration between The University of Chicago,
+Argonne National Laboratory and the University of Illinois. The code is
+released under the Apache 2.0 License and is available at GitHub. Parsl is a
+workflow system that can manage the data movement and execution of tasks on
+different hardware such as cloud, clusters and other resources. Tasks can be
+executed in parallel and will only be executed when required data is ready. The
+tasks that are executed are either bash scripts or Python functions. The
+functions will take arguments that are supplied during the definition of the
+workflow, such as parameters or outputs from previously executed tasks. The
+workflow system uses futures to setup the tasks to be executed, and will only
```

Comment:
Is "futures" a PARSL term?   Need some kind of text format?

Reply:
A future is a proxy for a result that is not yet available in concurrent programming languages. It seems to be rather general term, see e.g. relevant Wikipedia [entry](https://en.wikipedia.org/wiki/Futures_and_promises). According to its own [documentation](https://parsl.readthedocs.io/en/latest/userguide/futures.html#futures) Parsl uses them to emulate asynchronous code execution. To make the meaning explicit I added a link to Parsl's docs.